internal/pkg/providers/aws/aws.go

```diff
@@ -0,0 +1,77 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/stakater/Scaler/internal/pkg/cmd/common"
+)
+
+// Aws provider class implementing the Provider interface
+type Aws struct {
+	scalerName      *string
+	roleArn         string
+	region          *string
+	maxSize         *int64
+	minSize         *int64
+	desiredCapacity *int64
+}
+
+//GetName - Returns name of Aws Provider
+func (a *Aws) GetName() string {
+	return "Amazon Web Services"
+}
+
+// Init initializes the Aws Provider Configuration like Access Token and Reion
+func (a *Aws) Init(scalerOptions *common.ScalerOptions) error {
+	a.scalerName = &scalerOptions.ScalerName
+	a.roleArn = scalerOptions.RoleArn
+	a.region = &scalerOptions.Region
+	a.maxSize = &scalerOptions.Max
+	a.minSize = &scalerOptions.Min
+	a.desiredCapacity = &scalerOptions.Desired
+	return nil
+}
+
+// Scale - Modifies auto scaling group.
+func (a *Aws) Scale() error {
+
+	// Initial credentials loaded from SDK's default credential chain. Such as
+	// the environment, shared credentials (~/.aws/credentials), or EC2 Instance
+	// Role. These credentials will be used to to make the STS Assume Role API.
```

Comment:
@hamza3202 it must support working with Role; and we should completely discourage using credentials; credentials can be used for local some sort of "local profile"